components/layout/display_list/stacking_context.rs

```diff
@@ -1026,6 +1026,7 @@ impl BoxFragment {
         // > If a transform function causes the current transformation matrix of an object
         // > to be non-invertible, the object and its content do not get displayed.
         if !reference_frame_data.transform.is_invertible() {
+            self.clear_spatial_tree_node_including_descendants();
```

Comment:
The crash was happening because the spatial_tree_node wasn't set.
So in what case is it necessary to clear it?

Reply:
Clearing it is important so that behavior is consistent in the case that descendant `Fragment`s are retained. I think this can happen if the transformation changes and then we do a stacking context-only layout.

Reply:
Ah, sure, like changing from `scale(1)` to `scale(0)`.